pkgs/development/python-modules/home-assistant-chip-wheels/default.nix

```diff
@@ -0,0 +1,305 @@
+  # gn calls pigweed to read a dozen different files to generate
+  # a file looking like requirements.txt. It then calls pip
+  # to install this computed list of dependencies into a virtualenv.
+  # Of course, pip fails in the sandbox, because it cannot download
+  # the python packages.
+  # The documented way of doing offline builds is to create a folder
+  # with wheel files for all dependencies and point pip to it
+  # via its configuration file or environment variables.
+  # https://pigweed.dev/python_build.html#installing-offline
+  # The wheel of a python package foo is available as foo.dist.
+  # So that would be easy, but we also need wheels for transitive dependencies.
+  # the function saturateDependencies below computes this transitive closure.
+  #
+  # yes this list of dependencies contains both build tools and proper dependencies.
+  env.PIP_NO_INDEX = "1";
+  env.PIP_FIND_LINKS =
+    let
+      dependencies = [
+        appdirs
+        appnope
+        black
+        build
+        colorama
+        coloredlogs
+        coverage
+        click
+        cryptography
+        diskcache
+        googleapis-common-protos
+        google-cloud-storage
+        ipython
+        jinja2
+        json5
+        jsonschema
+        lark
+        mobly
+        mypy
+        mypy-extensions
+        mypy-protobuf
+        packaging
+        parameterized
+        pip-tools
+        prompt-toolkit
+        protobuf
+        psutil
+        ptpython
+        pyelftools
+        pygments
+        pykwalify
+        pylint
+        pyperclip
+        pyserial
+        python-daemon
+        pyyaml
+        requests
+        setuptools
+        six
+        sphinx
+        sphinx-argparse
+        sphinx-design
+        stringcase
+        toml
+        tornado
+        types-protobuf
+        types-pyyaml
+        types-requests
+        types-setuptools
+        watchdog
+        websockets
+        wheel
+        yapf
+      ];
+      depListToAttrs =
+        list:
+        builtins.listToAttrs (
+          map (dep: {
+            name = dep.name;
+            value = dep;
+          }) list
+        );
+      saturateDependencies =
+        deps:
+        let
+          before = deps;
+          new = lib.mergeAttrsList (
+            map (dep: depListToAttrs (dep.propagatedBuildInputs or [ ])) (lib.attrValues before)
+          );
+          after = before // new;
+        in
+        if lib.attrNames before != lib.attrNames after then saturateDependencies after else before;
+      saturateDependencyList = list: lib.attrValues (saturateDependencies (depListToAttrs list));
+      saturatedDependencyList = lib.filter (drv: drv ? dist) (saturateDependencyList dependencies);
+    in
+    lib.concatMapStringsSep " " (dep: "file://${dep.dist}") saturatedDependencyList;
+
+  configurePhase =
+    let
+      configureArguments = [
+        ''chip_project_config_include_dirs=["//.."]''
+        ''chip_crypto="boringssl"''
```

Comment:
Non-blocking fyi comment: I made this change to use boringssl, since upstream was building on a container image provided by upstream that was running old ubuntu with unsupported versions of openssl.

I don't think it's necessary now that we are building with an up-to-date toolchain, so we could substitute this for openssl if we want to.